### DISCUSSION TITLE
질문하나 있습니다!

### DISCUSSION THREAD

**Discussion by yunmo-k** [tedilabs/fastcampus-devops#34] [Q&A] [answered]
안녕하세요! 현재 devops 초격차 패키지 중 IaC관련 강의를 듣고 있는 수강생입니다. 한 가지 질문드릴것이 있습니다.

CH02_04 테라폼 HCL resource와 data 강의 中 7분 12초경 main.tf 에서 aws_instance 리소스 안에서
왜 data.aws_ami.ubuntu.id에서 data.aws_ami.ubuntu.image_id 로 바꾸셨는지 궁금합니다

**Comment by posquit0** [comment, marked as answer]
안녕하세요!
`data.aws_ami` 데이터 소스에 대한 문서는 다음과 같습니다.

- [Terraform Registry Docs: aws.data-sources.ami](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/ami)

Output Parameters의 정보는 문서에서 다음과 같이 확인하실 수 있는데, `data.aws_ami.*.image_id`와 `data.aws_ami.*.id`는 동일하다고 설명하고 있습니다.

```
image_id - The ID of the AMI. Should be the same as the resource id.
```

테라폼을 자주 사용하더라도 다루는 리소스와 데이터소스 종류가 많다보니 외워서 하기보다는 항상 문서를 확인해가며 작업을 하게 됩니다. 그러다보니 강의 초반에 `id`를 사용했던 것을 그새 잊고... 또 다시 문서를 보다가 `image_id`를 사용한 것으로 확인 됩니다. ㅎㅎ

별 이유는 없고, 둘 모두 같은 값이라서 정상 동작합니다!

**Reply by yunmo-k** [reply]
빠른 답변 감사드립니다! 한가지 추가질문이 있는데... 
강사님께서 간혹 nvim ~/ . terraformrc(예시) 명령어를 쳤을때  폴더가 열리는 것이 아닌 아래와 같은 리스트 목록이 뜹니다.
vi 명령어를 사용할 때 경로를 통해서 특정 파일을 접근하려고해서 그런것 같은데, 왜 이러는지 알 수 있을까요..? 
이런 경우에는 어떻게 해결해야하는지 궁금합니다

" ============================================================================
" Netrw Directory Listing                                        (netrw v165)
"   /home/ymk961028
"   Sorted by      name
"   Sort sequence: [\/]$,\<core\%(\.\d\+\)\=\>,\.h$,\.c$,\.cpp$,\~\=\*$,*,\.o$,\.obj$,\.info$,\.swp$,\.bak$,\~$
"   Quick Help: <F1>:help  -:go up dir  D:delete  R:rename  s:sort-by  x:special
" ==============================================================================
../                                                                                                                     ./
.ansible/
.aws/
.cache/
.config/
.landscape/
.local/
.ssh/
.terraform.d/
.vim/
01-start/
aws/

**Reply by yunmo-k** [reply]
아.. 설마 vi ~/ 입력 후 . 을 띄어쓰기 해서 vi ~/까지 인식하여 netbrw가 켜지는것인지요..?
그렇다면 vi ~/.terraformrc 이런 형식으로 들어가도 terraformrc 파일있는 곳으로 경로통해서가는것이 맞는것인지 궁금합니다

**Reply by posquit0** [reply]
네 맞습니다~ 홈디렉토리 `~/` 상에 위치한 `.terraformrc` 파일을 여는 것이라서 `vi ~/.terraformrc`라고 명령어를 입력하시면 됩니다.

**Comment by yunmo-k** [comment]
계속 질문해서 죄송합니다.. 마지막으로 
중간부터인가
There are some problems with the CLI configuration: 
 Error: The specified plugin cache dir /home/{사용자}/.terraform.d/plugin-cache cannot be opened: stat /home/{사용자}/.terraform.d/plugin-cache: no such file or directory
라는 오류가 계속 뜨는데 계속 찾아봐도 명쾌한 해결책이 안뜨네요.. 어떠한식으로 해결하면될런지요

**Reply by posquit0** [reply]
`~/.terraformrc`에서 플러그인 캐시와 관련된 설정을 하신 것 같아요. 그런데 `~/.terraform.d/plugin-cache` 디렉토리가 존재하지 않아 동작에 문제가 있는 듯 합니다.
`mkdir -p ~/.terraform.d/plugin-cache` 명령어 실행하여 디렉토리 생성 후 다시 시도 해보시기 바랍니다.

**Reply by yunmo-k** [reply]
네! 감사합니다. 좋은 하루 되십시오!
